src/pyseekdb/client/collection.py

```
@@ -385,6 +388,10 @@ def query(
                 n_results=10
             )
         """
+        # Validate _source parameter type
+        if _source is not None and not isinstance(_source, list):
+            raise TypeError(f"_source must be a list of strings or None, got {type(_source).__name__}")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)